src/FX.GrpcService/Services/FXService.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+using Grpc.Core;
+
+using GrpcModels;
+
+namespace GrpcService.Services
+{
+    public class FXService : FXGrpcService.FXGrpcServiceBase
+    {
+        public override async Task<GenericOutputParam> GenericMethod(GenericInputParam request, ServerCallContext context)
+        {
+            Console.WriteLine($"Received {request.MsgIn}");
+
+            var (type, _version) = Marshaller.ExtractMetadata(request.MsgIn);
+
+            var deserializedRequest = Marshaller.DeserializeAbstract(request.MsgIn, type);
+
+            if (deserializedRequest is Message message)
+            {
+                return await Task.FromResult(new GenericOutputParam { MsgOut = "received " + message.Text });
+            }
+            else if (deserializedRequest is LimitOrder limitOrder)
+            {
+                throw new NotImplementedException();
+            }
+            else if (deserializedRequest is MarketOrder marketOrder)
+            {
+                throw new NotImplementedException();
```

Comment:
@webwarrior-ws was the WebSocket proj throwing this before?

Reply:
replaced with implementations from Websocket project